.github/workflows/auto-recoder.yml

```diff
@@ -0,0 +1,41 @@
+name: auto-recoder
+on: 
+  pull_request:
+permissions:
+  contents: write
+#defaults:
+#  run:
+#    working-directory: scripts
+
+jobs:
+  auto-recode:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ref: ${{ github.head_ref }}
+    - name: a
+      run: pwd
+    - name: b
+      run: ls -al
+    - name: c
+      working-directory: scripts
+      run: ls -al
+    - name: d
+      working-directory: changes
+      run: ls -al
+    - name: e
+      working-directory: .github/workflows
+      run: ls -al
+    - name: run do work
+      working-directory: scripts
+      run: python dowork.py ${{github.event.pull_request.number}} ${{github.event.pull_request.title}}
```

Comment:
How about passing a raw python script? You may run some python code directly (without a file):
```bash
$ python -c """
import sys
sys.stdout.write('hello\n')
"""

>> hello
```